eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -66,30 +67,103 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
      */
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // Selection method, 0 = browse, 1 = dropdown
-        $storageDef->dataInt1 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionMethod'];
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        $doc = new DOMDocument('1.0', 'utf-8');
+        $root = $doc->createElement('related-objects');
+        $doc->appendChild($root);
 
-        // Selection root, location ID
-        $storageDef->dataInt2 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionRoot'];
+        $constraints = $doc->createElement('constraints');
+        if (!empty($fieldSettings['selectionContentTypes'])) {
+            foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
```

Comment:
I was suggesting something like:

```php
public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
{
    // [...]

    $root->appendChild($this->mapConstraintsToDOM($doc, $fieldSettings, ));
    $root->appendChild($this->mapSelectionTypeToDOM($doc, $fieldSettings));
    $root->appendChild($this->mapDefaultSelectionToDOM($doc, $fieldSettings));
}

private function mapConstraintsToDOM(DOMDocument $doc, array $fieldSettings)
{
    $constraints = $doc->createElement('constraints');

    if (!empty($fieldSettings['selectionContentTypes'])) {
        foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
            $allowedClass = $doc->createElement('allowed-class');
            $allowedClass->setAttribute('contentclass-identifier', $typeIdentifier);
            $constraints->appendChild($allowedClass);
            unset($allowedClass);
        }
    }

    return $constraints;
}

private function mapSelectionTypeToDOM(DOMDocument $doc, array $fieldSettings)
{
    $selectionType = $doc->createElement('selection_type');
    if (isset($fieldSettings['selectionMethod'])) {
        $selectionType->setAttribute('value', (int)$fieldSettings['selectionMethod']);
    } else {
        $selectionType->setAttribute('value', 0);
    }

    return $selectionType;
}

private function mapDefaultSelectionToDOM(DOMDocument $doc, array $fieldSettings)
{
    $defaultLocation = $doc->createElement('contentobject-placement');
    if (!empty($fieldSettings['selectionRoot'])) {
        $defaultLocation->setAttribute('node-id', (int)$fieldSettings['selectionRoot']);
    }
    
    return $defaultLocation;
}
```

And the mirror methods for DOM -> storage.
But it's not a hard requirement.